src/exporter/DaemonMetricCollector.cc

```diff
@@ -306,13 +306,15 @@ DaemonMetricCollector::get_labels_and_metric_name(std::string daemon_name,
     labels["instance_id"] = quote(tmp);
   }
   else if (daemon_name.find("rgw") != std::string::npos) {
-    // fetch intance_id for e.g. "okbvtv" from daemon_name=rgw.foo.ceph-node-00.okbvtv 
-    size_t pos = daemon_name.find_last_of(".");
```

Comment:
you can use [find_nth](https://www.boost.org/doc/libs/1_54_0/doc/html/boost/algorithm/find_nth.html) to get the third 4th element.

Reply:
Good simplification. It seems a good idea :-).